### ISSUE TITLE
refmt: "Encountered impossible type specification" error in valid private extensible variant type definition

### ISSUE THREAD

**Issue by jchavarri** [reasonml/reason#2302]
Input (OCaml):

```ocaml
module type S = sig
  type t = private ..
  type t += Foo
end
```

Output:
```
refmt: internal error, uncaught exception:
       Reason_pprint_ast.NotPossible("Encountered impossible type specification")
```

Expected output:
```reason
module type S = {
  type t = pri ..;
  type t +=
    | Foo;
};
```

Related: https://github.com/ocaml/ocaml/pull/1253

This was added in [4.06.0](https://github.com/ocaml/ocaml/blob/ebdad4fb7412293c87111c998e179bd461d9ab40/Changes#L1446).

**Comment by anmonteiro** [comment]
Probably fixed in #2036? 

The parser is currently written for the 4.04.2 AST

**Comment by anmonteiro** [comment]
seems fixed
